pyGHDL/cli/dom.py

```diff
@@ -0,0 +1,374 @@
+        )
+
+    # ============================================================================
+    # Common commands
+    # ============================================================================
+    # common arguments valid for all commands
+    # ----------------------------------------------------------------------------
+    @CommonSwitchArgumentAttribute(
+        "-d", "--debug", dest="debug", help="Enable debug mode."
+    )
+    @CommonSwitchArgumentAttribute(
+        "-v", "--verbose", dest="verbose", help="Print out detailed messages."
+    )
+    @CommonSwitchArgumentAttribute(
+        "-q", "--quiet", dest="quiet", help="Reduce messages to a minimum."
+    )
+    def Run(self):
+        ArgParseMixin.Run(self)
+
+    @DefaultAttribute()
+    def HandleDefault(self, _):
+        self.PrintHeadline()
+        self.MainParser.print_help()
+
+        self.WriteNormal("")
+        self.exit()
+
+    # ----------------------------------------------------------------------------
+    # create the sub-parser for the "help" command
+    # ----------------------------------------------------------------------------
+    @CommandAttribute("help", help="Display help page(s) for the given command name.")
+    @ArgumentAttribute(
+        metavar="Command",
+        dest="Command",
+        type=str,
+        nargs="?",
+        help="Print help page(s) for a command.",
+    )
+    def HandleHelp(self, args):
+        self.PrintHeadline()
+
+        if args.Command is None:
+            self.MainParser.print_help()
+        elif args.Command == "help":
+            self.WriteError("This is a recursion ...")
+        else:
+            try:
+                self.SubParsers[args.Command].print_help()
+            except KeyError:
+                self.WriteError("Command {0} is unknown.".format(args.Command))
+
+        self.WriteNormal("")
+        self.exit()
+
+    # ----------------------------------------------------------------------------
+    # create the sub-parser for the "version" command
+    # ----------------------------------------------------------------------------
+    @CommandAttribute("version", help="Display tool and version information.")
+    def HandleInfo(self, args):
+        self.PrintHeadline()
+
+        copyrights = __copyright__.split("\n", 1)
+        self.WriteNormal("Copyright:  {0}".format(copyrights[0]))
+        for copyright in copyrights[1:]:
+            self.WriteNormal("            {0}".format(copyright))
+        self.WriteNormal("License:    {0}".format(__license__))
+        authors = __author__.split(", ")
+        self.WriteNormal("Authors:    {0}".format(authors[0]))
+        for author in authors[1:]:
+            self.WriteNormal("            {0}".format(author))
+        self.WriteNormal("Version:    {0}".format(__version__))
+        self.exit()
+
+    # ----------------------------------------------------------------------------
+    # create the sub-parser for the "token-stream" command
```

Comment:
Is that "token-stream" correct or a pasto?

Reply:
You're right, that's pesto (from pyVHDLParser 😄)